syft/workers/message_handler.py

```diff
@@ -245,7 +245,13 @@ def respond_to_search(self, msg: SearchMessage) -> List[PointerTensor]:
             # decision to decide when to delete the tensor.
             ptr = obj.create_pointer(
                 garbage_collect_data=False, owner=sy.local_worker, tags=obj.tags
-            ).wrap()
+            )  # .wrap()
+
+            # Wrap only if the pointer points to a tensor.
+            # If it points to a generic object, do not wrap.
+            if isinstance(obj, PointerTensor):
+                ptr.wrap()
```

Comment:
Q: if you remove this if completely are the tests failing?

Reply:
Just pushed that to test

Reply:
Yeah...it is failling :(, it was ok before :+1:

Reply:
I reverted my changes and it still giving an error :O

Reply:
@AlanAboudib it seemed to be an intermittent and was not the result of your change.It was happening because of timeout when waiting for an async response. I re-ran the tests and all tests passed. You can make your change and we can debug further if the issue happens again.

Reply:
Thank you @shubham3121, I will push the changes then

Reply:
I think @shubham3121 referred to the changes with
```
    # Wrap only if the pointer points to a tensor.
            # If it points to a generic object, do not wrap.
            if isinstance(obj, PointerTensor):
                ptr.wrap()
```

Reply:
I was trying your idea of just remove `.wrap()` without adding my code. I just push my changes back now

Reply:
@AlanAboudib , Ques: You want to wrap Tensor objects (not necessarily PointerTensors objects) and not non Tensor objects, is that the behaviour you're looking for ?? 
If so, then the check should be `isinstance(obj, torch.Tensor)`. Because the currently the test is failing because the obj is an Tensor instance and not a PointerTensor object, as a result we are not wrapping it and hence results in error of doing .shape.

Reply:
Oh, my bad, I actually wanted to type:

```
if isinstance(ptr, PointerTensor):
```
Your solution works too. I will do

Thank you so much

Reply:
@AlanAboudib `ptr.wrap()` should ` prt = ptr.wrap()` as it is not an inline operation.